File: csharp/剑指50第一个只出现一次的字符.cs

public partial class Solution
{
    public char FirstUniqChar(string s)
    {
        Dictionary<char, int> dict = new Dictionary<char, int>();

        foreach(char ch in s)
        {
            dict[ch] = dict.ContainsKey(ch)?dict[ch] + 1 :1;
        }

        foreach(char key in dict.Keys)
        {
            if(dict[key]==1)
                return key;
        }
        return ' ';
    }
}

File: python/main/剑指50第一个只出现一次的字符.py
def firstUniqChar(s: str) -> str:
    counter = dict()

    for ch in s:
        if ch not in counter.keys():
            counter[ch] = 1
        else:
            counter[ch] += 1

    for k in counter:
        if counter[k] == 1:
            return k

    return " "


print(firstUniqChar("lleetcode"))
